### PR TITLE
Catalog of my things: Create game and author classes

### LINKED ISSUE #21: [1pt] Create a Game class. (Team member #3)
Create a Game class in a separate .rb file.

### LINKED ISSUE #23: [0.5pts] Create an Author class. (Team member #3)
Create an Author class with an association to the Item class (in a separate .rb file).

### LINKED ISSUE #24: [1pt] Define Game class properties in the construction method (Team member #3)
All Game class properties visible in the diagram should be defined and set up in the constructor method.

### LINKED ISSUE #26: [1pt] Define Author class properties in the construction method (Team member #3)
All Author class properties visible in the diagram should be defined and set up in the constructor method.

### LINKED ISSUE #28: [1pt] Implement methods (Team member #3)
- Implement methods:
    - add_item method in the Author class
        - should take an instance of the Item class as an input
        - should add the input item to the collection of items
        - should add self as a property of the item object (by using the correct setter from the item object)
    - can_be_archived?() in the Game class
        - should override the method from the parent class
        - should return true if parent's method returns true AND if last_played_at is older than 2 years
        - otherwise, it should return false

### PR DESCRIPTION
**Pull request by helio3197**
### Summary:
- Create Author class with the next attributes and methods:
  - first_name: string
  - last_name: string
  - id: int
  - items: arr(item)
  - add_item: method
- Create Game sub-class from Item parent class with the next attr and methods:
  - multiplayer: string
  - last_played_at: date
  - publish_date: date
  - can_be_archived?: method

### CODE REVIEW

File: author.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  def initialize(first_name, last_name, id = nil)
+    @first_name = first_name
+    @last_name = last_name
+    @id = id || rand(1..1000)
+    @items = []
+  end
+
+  def add_item(item)
+    @items << item
+    item.genre = self
```
Comment (estebanmual) [inline author.rb]: As you are creating the author class, try by modifying line 13 into `item.add_author(self)`

**Review by estebanmual — CHANGES_REQUESTED** [review]
# Status: Required Changes 🛑 

### Hi @helio3197 ! 👋🏻

You have made an excellent job so far! But there is an issue that you still need to work on to go to the next project but you are almost there!

File: author.rb (hunk 1 above)
Comment (helio3197) [reply author.rb]: Thanks, @estebanmual, but doing that will cause an infinite method call loop, I think we should refactor that line as `item.author = self` and modify the add_* methods from the Item class for removing the prop assignation and leave it only to be in the Author, Genre, Label classes.

**Review by estebanmual — APPROVED** [review]
# Status: Approved 🟢 

### Hi @helio3197 ! 👋🏻 

Great job implementing the changes.
Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:

Congratulations! 🎉

Cheers and Happy coding!👏👏👏



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**